src/main/java/seedu/address/logic/commands/MarkPaidCommand.java

```diff
@@ -49,13 +50,23 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         Person personToMarkPayment = lastShownList.get(targetIndex.getZeroBased());
+
+        List<Participation> participationList = personToMarkPayment.getParticipation();
+        // UnEnroll student from current participation first
+        for (Participation p: participationList) {
+            new UnEnrollCommand(targetIndex, p.getTutorialSubject()).execute(model);
```

Comment:
This part as well, the for loop will not work in this case too, so we should use a duplicated arraylist of participation objects to loop over instead.